journal-writer/src/main/scala/akka/stream/alpakka/persistence/writer/package.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.persistence
+
+import akka.actor.{ ActorContext, ActorRef }
+import akka.persistence.query.{ EventEnvelope, EventEnvelope2 }
+import akka.persistence.{ AtomicWrite, PersistentRepr }
+
+import scala.collection.immutable.{ Map, Seq }
+import scala.reflect.ClassTag
+
+package object writer {
+  val unsupported: Throwable = new IllegalArgumentException("JournalWriter only accepts EventEnvelope, immutable.Seq[EventEnvelope], EventEnvelope2 and immutable.Seq[EventEnvelope2]")
+
+  def replyWithFailure(replyTo: ActorRef, cause: Throwable = unsupported)(implicit ctx: ActorContext, self: ActorRef): Unit = {
+    replyTo ! akka.actor.Status.Failure(cause)
+    ctx.stop(self)
+  }
+
+  def replyWithSuccess(replyTo: ActorRef, msg: String = "ack")(implicit ctx: ActorContext, self: ActorRef): Unit = {
```

Comment:
package objects are visible to all classes in the package, there is nothing hidden about that. When using a structure where each package contains a single component like the JournalWriter, all functionality in the package object belongs to one component with one goal, writing to the journal. 

Given the component-per-package structure, the value I see in using package objects for methods and functions is that it makes classes more readable, focussing on the core functionality and not distracting you with details. When you want to see the details, look at the contents of the method. This lets you focus on the essentials. 

Also the code uses an architectural evident style in the naming conventions where each method speaks for himself. Think of it as zooming out on Google Maps, looking at cities (classes) and zooming in seeing roads and houses (clicking on methods).